indico/modules/events/papers/client/js/components/RevisionComment.jsx

```diff
@@ -0,0 +1,113 @@
+
+import {serializeDate} from 'indico/utils/date';
+import {Param, Translate} from 'indico/react/i18n';
+
+import {deleteComment} from '../actions';
+import {getPaperContribution, getPaperEvent, isDeletingComment} from '../selectors';
+import UserAvatar from './UserAvatar';
+
+export default function RevisionComment({comment, revision}) {
+  const dispatch = useDispatch();
+  const [confirmOpen, setConfirmOpen] = useState(false);
+  const {id: eventId} = useSelector(getPaperEvent);
+  const {id: contributionId} = useSelector(getPaperContribution);
+  const isDeletingCommentInProgress = useSelector(isDeletingComment);
+
+  return (
+    <div className="i-timeline-item">
+      <UserAvatar user={comment.user} />
+      <div className="flexrow i-timeline-item-content">
+        <div className="i-timeline-item-box header-indicator-left">
+          <div className="i-box-header flexrow">
+            <div className="f-self-stretch">
+              <Translate>
+                <Param name="userName" value={comment.user.fullName} wrapper={<strong />} /> left a
+                comment
+              </Translate>{' '}
+              {comment.visibility !== 'contributors' && (
+                <i
+                  className={`review-comment-visibility ${comment.visibility.name} icon-shield`}
+                  title={comment.visibility.title}
+                />
+              )}{' '}
+              <time dateTime={serializeDate(comment.createdDt, moment.HTML5_FMT.DATETIME_LOCAL)}>
+                {serializeDate(comment.createdDt, 'LL')}
+              </time>
+              {comment.modifiedDt && (
+                <span
+                  className="review-comment-edited"
+                  title={Translate.string('On {modificationDate} by {modifiedBy}', {
+                    modificationDate: serializeDate(comment.modifiedDt, 'LL'),
+                    modifiedBy: comment.modifiedBy.fullName,
+                  })}
+                >
+                  {' '}
+                  · <Translate>edited</Translate>
+                </span>
+              )}
+            </div>
+            {comment.canEdit && (
+              <div className="review-comment-action hide-if-locked">
+                <a
+                  href="#"
+                  className="i-link icon-edit js-edit-comment"
+                  title={Translate.string('Edit comment')}
+                />{' '}
+                <a
+                  onClick={() => setConfirmOpen(true)}
+                  className="i-link icon-cross js-delete-comment"
+                  title={Translate.string('Remove comment')}
+                />
+                <Confirm
+                  size="tiny"
+                  header={Translate.string('Remove comment')}
+                  open={confirmOpen}
+                  content={Translate.string('Are you sure you want to remove this comment?')}
+                  onCancel={() => setConfirmOpen(false)}
+                  onConfirm={() => {
```

Comment:
Does the Confirm auto-close? If not, what happens if the request is slow and you click twice?

Reply:
Discussed in person